### ISSUE TITLE
Macroexpansion with lazy seqs occurs in the wrong ns

### ISSUE THREAD

**Issue by yuhan0** [clojure-emacs/cider-nrepl#886]
## Expected behavior
Cider-nrepl's `macroexpansion` op should produce the same output as manually calling `macroexpand`.

## Actual behavior
If the macro definition contains lazily evaluated seqs, their contents will be evaluated with the wrong dynamic *ns* bindings (in the `user` ns), which can result in incorrect / misleading expansions.

## Steps to reproduce the problem
```clj
(ns repro)

(defmacro lazy []
  (cons (ns-name *ns*)
        (lazy-seq [(ns-name *ns*)])))

(macroexpand '(lazy))
;; => (repro repro)

(lazy) ;; <- M-x cider-macroexpand: (repro user)
```

### cider-nrepl version
0.49.1

### Java version

21.0.1

### Operating system

MacOS 14

**Comment by yuhan0** [comment]
Note: working on the fix, will submit a PR soon

**Comment by alexander-yakushev** [comment]
Fixed in #894.
